ga4gh/datamodel/datasets.py

```diff
@@ -227,6 +226,35 @@ def __init__(self, localId, dataDir, dataRepository):
                     self, localId, bamPath, dataRepository)
                 self.addReadGroupSet(readGroupSet)
 
+    def _metadataFileExists(self, fname):
+        """
+        Check for existence of description JSON.
+        """
+        return os.path.exists(fname)
+
+    def _parseMetadataJson(self, str):
+        """
+        Parse metadata JSON into python dict.
+        """
+        return json.loads(str)
+
+    def _getMetadataFilename(self, path, localId):
+        """
+        Given a path and identifier create a path
+        to the expected metadata JSON.
+        """
+        return os.path.join(path, localId + ".json")
+
+    def _isAnnotated(self, path, localId):
```

Comment:
Shouldn't this be a public function of the VariantSet? Then we can have
```python
                 variantSet = variants.HtslibVariantSet(
                      self, localId, relativePath, dataRepository)
                 self.addVariantSet(variantSet)
                 if variantSet.isAnnotated():
                      variantAnnotationSet = variants.HtslibVariantAnnotationSet(
                           self, localId, relativePath, dataRepository,  variantSet)
                      self.addVariantAnnotationSet(variantAnnotationSet)
```
Keeping a reference to the 'parent' variant set is useful, and will avoid having to reproduce methods in the variantAnnotationSet.